.github/workflows/run-tests.yml

```diff
@@ -492,7 +492,7 @@ jobs:
         uses: actions/upload-artifact@v4.4.3
         if: ${{ !cancelled() && !inputs.run_single_functional_test }}
         with:
-          name: junit-xml--${{github.run_id}}--${{github.run_attempt}}--${{matrix.runs-on}}--${{matrix.name}}--${{matrix.shard_index}}--functional-test
+          name: junit-xml--${{github.run_id}}--${{github.job_id}}--${{github.run_attempt}}--${{matrix.runs-on}}--${{matrix.name}}--${{matrix.shard_index}}--functional-test
```

Comment:
Looks like it should be `github.job`: https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/accessing-contextual-information-about-workflow-runs#github-context

Reply:
Actually, I'm realizing that this job ID is not the numeric identifier we are looking for. `github.job` is an identifier for the job _type_, whereas what we want is the numeric job ID used in the URL. For some reason GitHub doesn't make it easy to find that. However, on the plus side we do have a workflow that shows how to obtain this:

https://github.com/temporalio/saas-cicd/blob/11b3918efe21f4d4aafecb9631c428cd2d17b10c/.github/workflows/execute-oss-cicd-workflow.yaml#L126
https://github.com/temporalio/saas-cicd/blob/main/oss-cicd/internal/github/get-job-id-for-job-name

Reply:
so much magic...thanks Dan